### PR TITLE
Update Swagger for Static Web Apps database connections

### PR DESCRIPTION
**Pull request by joslinmicrosoft**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
@naveedaz I don't know why but I cannot add you to reviewer.

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:978]: >     "pattern": "^[a-zA-Z0-9]+$", [](http://example.com/codeflow?start=7&length=33)

are all the patterns you're adding exactly matching the rules you previously enforced?

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:1068]: I think technically this change would require a new api version. Was your server already returning it in practice though? Or is this an unreleased ap-version??

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:2292]: You might want to refactor this, to reduce copy/paste risk - put this in "Parameters" block and reference it everywhere with $ref.

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4711]: Don't files have some unique id (or tuple of ids) you can address them with... like a path?

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:995]: >StaticSiteDatabaseConnectionRequestPropertiesARMResource [](http://example.com/codeflow?start=37&length=56)

Possibly too late now.... (breaking change?) 
But I'd just hope to give it a name that's simpler like
'DatabaseConnectionRequest'

**Comment by TimLovellSmith** [comment]
>     }


Common parameters best practice - use common types
There are definitions of common parameters in common-types/v3/types.json that you can use by $ref instead of declaring your own, like SubscriptionIdParameter, OperationIdParameter, and ApiVersionParameter,
See
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L630](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L630)



---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:5375 in c73d7d9. [](commit_id = c73d7d9035fef18ef74ade17e5975fd5f66fe114, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "Identity to use for Key Vault Reference authentication.",

What format? Is it a resource id, or a guid, or what?

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4301 in c73d7d9. [](commit_id = c73d7d9035fef18ef74ade17e5975fd5f66fe114, deletion_comment = False)

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4348]: >StaticSiteDatabaseConnectionOverviewARMResource [](http://example.com/codeflow?start=35&length=47)

What's the difference between a 'StaticSiteDatabaseConnectionOverview' and a 'StaticSiteDatabaseConnectionOverviewARMResource'? Aren't they conceptually the same thing?

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4745]: >StaticSiteDatabaseConnectionPatchRequestPropertiesARMResource [](http://example.com/codeflow?start=5&length=61)

Why did you remove the allOf inheritance of ProxyResoruce?

**Review by TimLovellSmith — CHANGES_REQUESTED** [review]
:clock1:

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (joslinmicrosoft) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:978]: For `environmentName`, yes we have already been enforcing this but it was just not documented before. For `databaseConnectionName` this is a new API, but it will also be enforced this way.

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (joslinmicrosoft) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:1068]: So on this point I wanted to clarify a couple things

1. I'm not sure how to improve this, but I think the diff displayed here is a bit misleading. So far, I have only made additions to this file for the new database connections APIs (no removals); however, since they follow a similar format to other APIs, I think the diff tool is displaying it as replacing the user provided function app APIs with the database connection APIs and then also adding the user provided function app APIs. Let me know if this was your concern and if this clears anything up, I'm not intending on changing any of the existing APIs.

2. This 2022-09-01 api version I believe is a new, unreleased api version anyways. My PR is made against our broader team's feature branch.

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (joslinmicrosoft) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4711]: Can you help clarify what level of uniqueness this is supposed to be (I tried looking at the docs but couldn't find a clear answer)? Should it be unique within the `configurationFiles` array, or some sort of globally unique id across all customers? Like within a `configurationFiles` array, all the file names would be different, but these file names could be the same in the configurationFiles of another database connection.

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (joslinmicrosoft) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4745]: The ProxyOnlyResource definition contains the properties `name` and `type` and I was getting a flagged with this error: https://github.com/Azure/azure-openapi-validator/blob/main/docs/un-supported-patch-properties.md

**Comment by joslinmicrosoft** [comment]
> > ```
> > }
> > ```
> 
> Common parameters best practice - use common types There are definitions of common parameters in common-types/v3/types.json that you can use by $ref instead of declaring your own, like SubscriptionIdParameter, OperationIdParameter, and ApiVersionParameter, See
> 
> https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L630
> 
> Refers to: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:5375 in [c73d7d9](https://github.com/Azure/azure-rest-api-specs/commit/c73d7d9035fef18ef74ade17e5975fd5f66fe114). [](commit_id = [c73d7d9](https://github.com/Azure/azure-rest-api-specs/commit/c73d7d9035fef18ef74ade17e5975fd5f66fe114), deletion_comment = False)

@TimLovellSmith regarding this, as this pattern is used across our broader team I would need to defer to @naveedaz for how we would want to adress this.

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (joslinmicrosoft) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4348]: Can you clarify this comment? In the most recent iteration as of your review, I have removed references to StaticSiteDatabaseConnectionOverview in favor of StaticSiteDatabaseConnectionOverviewARMResource.

**Comment by joslinmicrosoft** [comment]
> > ```
> >       "description": "Identity to use for Key Vault Reference authentication.",
> > ```
> 
> What format? Is it a resource id, or a guid, or what?
> 
> Refers to: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4301 in [c73d7d9](https://github.com/Azure/azure-rest-api-specs/commit/c73d7d9035fef18ef74ade17e5975fd5f66fe114). [](commit_id = [c73d7d9](https://github.com/Azure/azure-rest-api-specs/commit/c73d7d9035fef18ef74ade17e5975fd5f66fe114), deletion_comment = False)

I've updated the description for `connectionIdentity` to clarify this. I am leaving existing APIs as is to limit the scope of this PR to just the addition of the database connection APIs

**Comment by joslinmicrosoft** [comment]
@TimLovellSmith thank you for the review, I've addressed some of your comments with pushed changes and others added some more discussion. I'm unable to remove the `ARMChangesRequested` tag, can you or maybe @weidongxu-microsoft help with that to continue with the review?

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json]: Oh OK. That seems weird. I mean, I thought they are all modeled as read-only, aren't they?

Comment (TimLovellSmith) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json]: It might have been a problem with the specific ProxyResource definition that you were using.

          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4626]: My impression is that, as long as its unique within this particular array, so that it can be used for key-value lookup, its OK... I'm not totally sure about this however.

[https://azure.github.io/autorest/extensions/#x-ms-identifiers](https://azure.github.io/autorest/extensions/#x-ms-identifiers)

Comment (TimLovellSmith) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4626]: @weidongxu-microsoft do you know?

Comment (weidongxu-microsoft) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4626]: I am not sure either. I had the impression that it is used by ARM or some of its subservice.

Same as Tim, I assume it to be unique for the item of the array. In this particular API, it maybe "fileName" (that one site won't have 2 configure override on same file?).

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (raosuhas) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:971]: >} [](http://example.com/codeflow?start=12&length=1)

Why do you need a separate object for request and response ? Ideally you should be having the same object in REST that is PUT and GET except for any secrets or readonly properties

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (raosuhas) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json]: I think this is fine cause you really should not be providing those top level properties for PATCH

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (raosuhas) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4676]: >connectionString [](http://example.com/codeflow?start=13&length=16)

x-ms-secret ?

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (raosuhas) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4613]: >true [](http://example.com/codeflow?start=26&length=4)

is this a secret ?
 [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret)

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (raosuhas) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4683]: I am a bit confused .. you have defined these as readonly in the PUT but you callow the user to update them on PATCH ?

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (raosuhas) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4713]: Can you please update the x-ms-secret and the readonly properties corectly everywhere for this PR : 

[https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-mutability](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-mutability)

Comment (TimLovellSmith) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4713]: +1 that. Its a breaking change with old api versions... for sake of security improvement, so it should be approved for a 1 year deprecation timeline

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (joslinmicrosoft) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:1033]: Could you clarify what exactly is a breaking change here? I am adding new APIs related to databaseConnections, which is a new set of APIs we'll be supporting. To clarify again, there's no change to the existing APIs, I believe it's just an artifact of the diffing tool (for example if you look at the API for `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/staticSites/{name}/builds/{environmentName}/userProvidedFunctionApps` which seems to have been removed, it actually appears around line 1380 in the new file).

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4611]: >secret [](http://example.com/codeflow?start=20&length=6)

+do we also need x-ms-mutability: "create, write" or is it implied by x-ms-secret? I forget what the tooling needs here.

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4588]: >ARM [](http://example.com/codeflow?start=54&length=3)

Think ARM as implementation detail... instead of an ARM resource, better to call it a resource, or an Azure resource, or even just say its "A static site database connection [that you can use to ...]" .

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4600]: >resourceId [](http://example.com/codeflow?start=13&length=10)

Naming; 'resourceId' isn't clear which resource its talking about. I'd suggest to call it 'databaseResourceId'

Comment (TimLovellSmith) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4600]: Are any parameters required?

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4616]: Naming: Would it be clearer to call it 'databaseLocation'? Since if you just see 'region' without context, you might assume its the region of the connection (whatever that would mean) or perhaps the static site?

Comment (TimLovellSmith) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4616]: Is this field mutable? Or set at create time? Or is it effectively readonly because the value is automatically determined by which database resource id you pass in?

Comment (joslinmicrosoft) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4619]: The field is indeed mutable.

Regarding the naming for `region` and `resourceId` we specifically went with these rather than prefixing with `database` because we found that redundant within database connection requests/responses (all the properties relate to the database). Let me know if this would be a blocking requirement, as we've already built out a lot of this feature with these names and it would be quite a bit of work to revise and redeploy.

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (joslinmicrosoft) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4600]: Updated required parameters, and am addressing the naming in a below comment

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4619]: @joslinmicrosoft They are just suggestions, as resourceId is in general 'reasonable' and in line with guidelines . I went with these suggestions partly because, although the naming may seem redundant today, it may _not_ seem as redundant or clear in future API versions if you add in more properties that reference more resources.

**Comment by joslinmicrosoft** [comment]
Hi @TimLovellSmith @raosuhas @weidongxu-microsoft I've addressed the comments and gotten the required checks to pass. Is there anything else pending or can we merge this PR?

**Comment by weidongxu-microsoft** [comment]
@joslinmicrosoft

I see Tim has a change request on the PR, so I will wait a bit for him to check again.

Do you have approval from Navy's team (owner of release-web-Microsoft.Web-2022-09-01 branch)?

I should be able to merge after above responded/solved.

**Comment by joslinmicrosoft** [comment]
@TimLovellSmith do you mind giving another review (since we're currently blocking on the requested changes)?

@weidongxu-microsoft I will check in with Navy, thanks!

**Comment by naveedaz** [comment]
@weidongxu-microsoft Approved from my side.

**Comment by weidongxu-microsoft** [comment]
Hi @TimLovellSmith, your change requests are either updated by service, or explained in details. Could you check again, that whether the PR is acceptable to you.

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:1080]: >DatabaseConnectionPatchRequest [](http://example.com/codeflow?start=37&length=30)

Can you just make the schema of the patch request 'DatabaseConnectionPatchRequest'? And add x-ms-mutability: "create,read" notations on any properties which aren't upddatable? OR does it really need to be different?

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:1334]: >StaticSites_GetBuildDatabaseConnectionsWithDetails [](http://example.com/codeflow?start=24&length=50)

nitpick, why put 'WithDetails' in the name? Aren't you just returning all the normal resource properties? Where does the WithoutDetails scenario come in?

**Comment by TimLovellSmith** [comment]
>           "description": "Identity to use for Key Vault Reference authentication.",

To be clear, I'm just asking you to be specific in the description, as I consider this not clear enough to understand without also reading the examples.

---
In reply to: [1382665064](https://github.com/Azure/azure-rest-api-specs/pull/22159#issuecomment-1382665064) [](http://example.com/codeflow?ancestors=1382665064,1382665064)

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4301 in c73d7d9. [](commit_id = c73d7d9035fef18ef74ade17e5975fd5f66fe114, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "Identity to use for Key Vault Reference authentication.",

I don't see this property in the examples in this PR BTW?

---
In reply to: [1432561416](https://github.com/Azure/azure-rest-api-specs/pull/22159#issuecomment-1432561416) [](http://example.com/codeflow?ancestors=1432561416,1382665064,1382665064)

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4301 in c73d7d9. [](commit_id = c73d7d9035fef18ef74ade17e5975fd5f66fe114, deletion_comment = False)

**Review by TimLovellSmith — APPROVED** [review]
:shipit:

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (TimLovellSmith) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4607]: >connectionString [](http://example.com/codeflow?start=13&length=16)

nitpick, I have a doubt that you can declare it as required AND not return it in responses, as required tends to get mapped to non-nullable...

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (joslinmicrosoft) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:1080]: I think I really do need a separate schema because of this discussion https://github.com/Azure/azure-rest-api-specs/pull/22159#discussion_r1080649625. Otherwise the ProxyOnlyResource inheritance causes issues for the PATCH requests.

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (joslinmicrosoft) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:1334]: This is mainly to distinguish the POST (which provides the connection string) from the regular GET (which does not provide the connection string). The `WithDetails` is to be a bit more generic for future-proofing if additional properties which should be in the POST response but not GET appear.

**Comment by joslinmicrosoft** [comment]
> > ```
> >       "description": "Identity to use for Key Vault Reference authentication.",
> > ```
> 
> To be clear, I'm just asking you to be specific in the description, as I consider this not clear enough to understand without also reading the examples.
> 
> In reply to: [1382665064](https://github.com/Azure/azure-rest-api-specs/pull/22159#issuecomment-1382665064)
> 
> Refers to: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json:4301 in [c73d7d9](https://github.com/Azure/azure-rest-api-specs/commit/c73d7d9035fef18ef74ade17e5975fd5f66fe114). [](commit_id = [c73d7d9](https://github.com/Azure/azure-rest-api-specs/commit/c73d7d9035fef18ef74ade17e5975fd5f66fe114), deletion_comment = False)

Gotcha, yes I understand the feedback but I'm saying that this is out of scope for this PR. The keyVaultReferenceIdentity is part of an existing API, I can update it in a separate PR but would like to keep this one focused to database connections APIs only if that's ok.

File: specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json
Comment (joslinmicrosoft) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json]: got it, pushed up a commit to remove `connectionString` from the required array

**Comment by joslinmicrosoft** [comment]
Thanks again @TimLovellSmith, @weidongxu-microsoft. Can you please give one more review and if all is good merge? Thanks!

**Comment by weidongxu-microsoft** [comment]
Oh, sorry I've merged it before Tim get a chance to provide more feedbacks.

@joslinmicrosoft 
Since the target is a feature branch, if Tim give more feedback that you think should be included, please raise a follow-up PR.

**Pull request merged**
